database/src/main/runs/_get_id_partitioning.sql

```diff
@@ -24,22 +24,22 @@ $$
 --      Gets the id of the provided segmentation, if it exits
 --
 -- Parameters:
---      i_segmentation      - segmentation to look for
+--      i_partitioning      - partitioning to look for
 --
 -- Returns:
---      key_segmentation    - id of the segmentation if it exists, NULL otherwise
+--      id_partitioning     - id of the segmentation if it exists, NULL otherwise
 --
 -------------------------------------------------------------------------------
 DECLARE
 BEGIN
-    SELECT SEG.id_segmentation
-    FROM runs.segmentations SEG
-    WHERE SEG.segmentation = i_segmentation
-    INTO key_segmentation;
+    SELECT PAR.id_partitioning
+    FROM runs.partitionings PAR
+    WHERE PAR.partitioning = i_partitioning
+    INTO id_partitioning;
 
     RETURN;
 END;
 $$
 LANGUAGE plpgsql VOLATILE SECURITY DEFINER;
 
-GRANT EXECUTE ON FUNCTION runs._get_key_segmentation(hstore) TO atum_user;
+ALTER FUNCTION runs._get_id_partitioning(JSONB) OWNER TO atum_owner;
```

Comment:
why not GRANT EXECUTE and why owner and not the user? I thought that only DB, schemas, and tables are owned by owner, and that the functions are supposed to be executed as the user, not owner

Reply:
or rather, is the idea here that these 'private' functions shouldn't be called by the user at all? Would it work if the user executed a function that calls this function?

Reply:
* yes the `_` at the beginning of the function name signs a "private" function.
* private function is not to be called outside, therefore not grants; thanks to that it can be lenient on input sanitation
* ownership automatically means execution right
* the magic of the function being (indirectly) usable by outside user is in the `SECURITY DEFINER` specification in the functions - it means that all the code within the function will be executed with right of the ROLE that owns the function not the ROLE that called the function. This prevents the needs to maintain cascade of access rights and opening _private_ functions to outside use.

Reply:
awesome, thanks a lot for such detailed explanation!